### ISSUE TITLE
Can the examples be relicensed under public domain?

### ISSUE THREAD

**Issue by t184256** [spatialaudio/jackclient-python#32]
Can the examples be relicensed under public domain? They are a good starting point, but the requirement of retaining the copyright notice in the work derived from them doesn't seem appropriate to me.

Or am I missing something?

**Comment by mgeier** [comment]
I'm open to your suggestion, but I don't know the legal details nor can I remember seeing something like that before.

Can you point me to examples where something like this is done?

I think it's very common for people to take code examples as starting point for their own projects, so probably this use case is already covered by the copyright laws?

Another aspect is that I modeled at least 2 examples (`thru_client.py` and `midi_sine.py`) after examples from JACK (which is GPL/LGPL?) ... so did I already violate their rights?

**Comment by t184256** [comment]
> Can you point me to examples where something like this is done?
Sorry, but I somehow fail to answer this question. I recall that it's common for examples to be in PD, but I can't find any guides on that. Maybe this is a non-issue after all.

> Another aspect is that I modeled at least 2 examples (thru_client.py and midi_sine.py) after examples from JACK (which is GPL/LGPL?) ... so did I already violate their rights?
No way. I fail to determine [thru_client's](https://github.com/jackaudio/jack2/blob/master/example-clients/thru_client.c) license (which makes me even more confused), but deriving a work from a specifically GPL'ed [midisine](https://github.com/jackaudio/jack2/blob/master/example-clients/midisine.c) and publishing it under GPL is clearly OK. So no, I don't think you've already violated anything.

**Comment by mgeier** [comment]
OK, if there are no examples and no guidelines, there's nothing I can do ...

python-sounddevice is licensed under the MIT license, so I guess I'm not allowed to include a derived work (or unchanged, for that matter) of "midisine", which is licensed under the GPL.

But the question is if it even counts as derived work, since I'm using none of the actual code.

The whole situation is not very clear to me, so if you have any more information, please speak up!

**Comment by mgeier** [comment]
I just had a look at the actual [license text of the MIT license](https://opensource.org/licenses/MIT). It says:

```
The above copyright notice and this permission notice shall be included in
all copies or substantial portions of the Software.
```

I would say that each of the examples is not a *substantial portion* of the software, therefore you are not obliged to include the license if you base your work off of one of the examples.

IANAL, but doesn't that sound reasonable?

**Comment by t184256** [comment]
> IANAL, but doesn't that sound reasonable?

Me too, I think it does. Actually I probably already got the answer for my specific case.

Probably the simplest thing that you can do is to trace the sources back to what have inspired them and simply use the same license for them if you don't have anything against that.
That would sure make the situation clearer and I can't imagine how could it violate anything if the licenses in question were MIT, GPL and the likes.

**Comment by mgeier** [comment]
As far as I understand, you should be careful when mixing MIT (and similar) with GPL.

Including MIT-licensed parts into a GPL-licensed project is, to my knowledge, no problem. Including GPL-licensed parts into an MIT-licensed project is problematic (or just forbidden?).

So I guess if the original JACK example is GPL-licensed, I cannot include it into my MIT-licensed code base. But I *don't want* to put my own examples under the GPL, because I want people to actually derive their own work from my software.

I think the JACK people actually didn't want to restrict the use of their examples, I guess they just included the license text out of habit.

Licensing my own examples under CC0 would send a clearer message, but I hesitate to do that without precedent.

**Comment by mgeier** [comment]
@t184256 Is there any new information available?

If not, I'll close this at some point.

**Comment by t184256** [comment]
I was not and am not knowledgeable in this area, sorry. Anyway, the answer 'no' is fine for me.

**Comment by mgeier** [comment]
The answer is not "no", but "not until I see a precedent".

Apart from that, use of the examples might fall under "fair use" or the examples might be too trivial to be copyright-able anyway, or ... we would have to ask a lawyer (or several of them).

For now, I will not change anything regarding the license, but rest assured that I personally will not sue you or anyone if they use the examples for whatever they want.

I'm closing this for now, but if any new information turns up, I'll reopen it.
